diff-containers/README.md

```diff
@@ -1 +1,238 @@
-# diff-containers
\ No newline at end of file
+# diff-containers
+
+This package implements difference datatypes for container types. Currently,
+there is only one such implemented difference datatype for `Data.Map.Strict`.
+
+## Differences on `Data.Map`s
+
+A difference datatype for maps is defined and implemented in the
+`Data.Map.Diff.Strict` module. A key property of our definition of diffs is that
+diffs can not only be combined/summed, they can also be inverted and summed
+together, which facilitates *subtraction* of diffs. Our definition of diffs is
+not only a `Semigroup` and a `Monoid`, it is also a `Group`.
+
+The goals for our definition of diffs are twofold:
+1. Provide definitions that adhere to the type class laws for `Semigroup`,
+   `Monoid` and `Group`.
+2. Applying diffs to maps should have a nice semantics.
```

Comment:
Hmmm, maybe we should elaborate on what a "nice semantics" is. Do we mean sound?

Reply:
Done